_episodes/23-continuous-integration-automated-testing.md

```diff
@@ -410,4 +410,26 @@ which potentially saves us a lot of time waiting for testing results.
 Overall, this approach allows us to massively scale our automated testing
 across platforms we wish to test.
 
+> ## Failed CI Builds
+> A CI build can fail when, e.g. a used Python package no longer supports a particular version of 
+> Python indicated in a GitHub Actions CI build matrix. In this case, the solution is either to 
+> downgrade the package version (and not use the latest one like we have been doing in this course),
+> or upgrade the Python version in the build matrix.
```

Comment:
Could add a comment on what the best solution is in this situation, i.e. upgrade where possible, as opposed to downgrade.

Reply:
Done!